### PR TITLE
task-arrays

### PR DESCRIPTION
**Pull request by MikhailBiltsevich**
Each function implements 1 subtask. Functions order is same as your Skype message

### CODE REVIEW

File: arrays.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+function copyArr(arr) {
+  return arr.map((value) => value);
```
Comment (AliakseiPuchyk) [inline arrays.js]: Oh, maybe it is my mistake, i didn't tell you: you shouldn't use default functions.
Could you please avoid using _map_ and _join_ functions for first two tasks?

File: arrays.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+function colonOdd(num) {
+  if (typeof num !== 'number') {
+    return undefined;
+  }
+
+  const arr = num.toString().split('');
+
+  for(let i = 1; i < arr.length; i += 1) {
```
Comment (AliakseiPuchyk) [inline arrays.js]: You can simply use increment here
`i += 1` => `i++`

Comment (AliakseiPuchyk) [inline arrays.js]: I think array start from 0 element. 😁
I understand why you decided to start from 1, but you can do the same things if you start from 0 element.

File: arrays.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+     const num = Number.parseInt(arr[i]);
+     const prevNum = Number.parseInt(arr[i - 1]);
+
+     if ((num * prevNum) % 2 !== 0) {
```
Comment (AliakseiPuchyk) [inline arrays.js]: 👍

File: arrays.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+       arr.splice(i, 0, ':');
+       i += 1;
```
Comment (AliakseiPuchyk) [inline arrays.js]: Again, you can use increment

File: arrays.js (hunk 2 above)
Comment (AliakseiPuchyk) [reply arrays.js]: It would be more clean

File: arrays.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
+function copyArr(arr) {
```
Comment (AliakseiPuchyk) [inline arrays.js]: Could you please use **arrow function** for all functions?

File: arrays.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+function removeDuplicates(arr) {
+  for (let i = 0; i < arr.length; i += 1) {
+    const currentItem = arr[i];
+    const hasDuplicates = arr.slice(i + 1).some(isDuplicate);
+
+    function isDuplicate(item) {
+      let isEqual = false;
+      if (typeof item === typeof currentItem) {
```
Comment (AliakseiPuchyk) [inline arrays.js]: If you using '===' i think you don't need to check type of element

File: arrays.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+    if (hasDuplicates) {
+      arr = arr.filter((item, index) => index <= i || !isDuplicate(item));
+    }
+  }
+
+  return arr;
+}
+
+function countIdentic(arr) {
```
Comment (AliakseiPuchyk) [inline arrays.js]: I think you don't correctly understood task 😓

`console.log(countIdentic([1, 1, 2, 2, 1, 1]))`  => Expected result:  2

File: arrays.js (hunk 1 above)
Comment (MikhailBiltsevich) [reply arrays.js]: sure

File: arrays.js (hunk 2 above)
Comment (MikhailBiltsevich) [reply arrays.js]: ok. When I was using airbnb-config It was requiring to use i += 1 instead i++

File: arrays.js (hunk 5 above)
Comment (MikhailBiltsevich) [reply arrays.js:1]: Should I use arrow functions for all functions?

File: arrays.js (hunk 4 above)
Comment (MikhailBiltsevich) [reply arrays.js]: ok

File: arrays.js (hunk 6 above)
Comment (MikhailBiltsevich) [reply arrays.js:36]: ok

Comment (MikhailBiltsevich) [reply arrays.js]: ok

File: arrays.js (hunk 2 above)
Comment (MikhailBiltsevich) [reply arrays.js]: ok

File: arrays.js (hunk 5 above)
Comment (MikhailBiltsevich) [reply arrays.js]: ok

**Review by AliakseiPuchyk — APPROVED** [review]

**Pull request merged**